### ISSUE TITLE
Tests are failing

### ISSUE THREAD

**Issue by fingolfin** [gap-packages/liepring#16]
At some point changes were made to this package that cause its test suite to fail. Some of these are harmless. E.g. I think printing changed, so now sometimes instead of `LiePRing of dimension 4 over prime p with parameters [ w ]` now ` <LiePRing of dimension 4 over prime p>` is printed.

But others are worrying. E.g. take this example from the manual:
```
gap> K := SpecialisePrimeOfLiePRing(L, 7);
<LiePRing of dimension 6 over prime 7>
gap> b := BasisOfLiePRing(K);
[ l1, l2, l3, l4, l5, l6 ]
gap> U := LiePSubring(L, [5*b[1]]);
<LiePRing of dimension 1 over prime p>
```
Re-running this yields an error in the last line: `Error, QuotRem: division case 2`

Moreover,
```
gap> grps := GroupsViaLiePRings(5,5);;
gap> Length(grps) = NrSmallGroups(5^5);
true
gap> Set(GroupsViaLiePRings(5,5),IdGroup) = List([1..77],i->[5^5,i]);
true
```
has changed and now reports `false` in both cases.

@beick any insights on these?

**Comment by beick** [comment]
Lieber Max,

ich habe nach einem Computer-Crash einen neuen Mac bekommen, habe auf diesem die neueste Version von
GAP installiert und das LiePRing Package ausprobiert. Zu Deinen Anmerkungen:

>At some point changes were made to this package that cause its test suite to fail. Some of these are harmless. 
>E.g. I think printing changed, so now sometimes instead of `LiePRing of dimension 4 over prime p with parameters [ w ]` now 
>`<LiePRing of dimension 4 over prime p>` is printed.

Hast Du ein konkretes Beispiel dafür?

>gap> K := SpecialisePrimeOfLiePRing(L, 7);
><LiePRing of dimension 6 over prime 7>
>gap> b := BasisOfLiePRing(K);
>[ l1, l2, l3, l4, l5, l6 ]
>gap> U := LiePSubring(L, [5*b[1]]);
><LiePRing of dimension 1 over prime p>

Ja, das ist ein Fehler im Manual. U sollte ein Subring von K sein, nicht von L.

>gap> grps := GroupsViaLiePRings(5,5);;
>gap> Length(grps) = NrSmallGroups(5^5);
>true
>gap> Set(GroupsViaLiePRings(5,5),IdGroup) = List([1..77],i->[5^5,i]);
>true
>
>has changed and now reports `false` in both cases.

Das kann ich nicht reproduzieren.

Ausserdem möchte ich das LiePRing Package erweitern um die LiePRings der
Ordnung pˆ8 von maximaler Klasse. 

Ich vermute, ich sollte die Development-Version von GAP per GitHub herunterladen, dort 
Änderungen machen und dann wieder pushen? Ist das der Dienstweg für solche Probleme?

Viele Grüsse,

Bettina

**Comment by fingolfin** [comment]
Dear Bettina,

I am not sure if you realize that you posted this on the GitHub issue tracker (by replying to the email notification -- emails are turned into issue comments... If you prefer, we can hide or delete your comments here (although I see nothing problematic or "secret" in them *shrug*).


I also noticed that you actually never *released* the new LiePRing version, so everyone still has 1.9.2, which might explain why e.g. @alex-konovalov did not notice these problems yet -- the package did not pass through the package distribution tests.

For a release, it is not sufficient to update the PackageInfo.g file; you need to use the ReleaseTools script; i.e., the [three steps described here](https://github.com/gap-system/ReleaseTools#the-release-process-short-version).

This would also have revealed that converting the manual to HTML is currently broken... I'll submit a fix for that shortly.

---

Regarding your questions/remarks: All I am talking about here can be seen by running the LiePRing test suite, i.e. by loading GAP with the right LiePRing installed, and then doing
```gap
Read(Filename(DirectoriesPackageLibrary("liepring", "tst"), "testall.g"));
```
Specifically:


> > At some point changes were made to this package that cause its test suite to fail. Some of these are harmless. E.g. I think printing changed, so now sometimes instead of `LiePRing of dimension 4 over prime p with parameters [ w ]` now `<LiePRing of dimension 4 over prime p>` is printed.
>
> Hast Du ein konkretes Beispiel dafür?

We used to get this in the tests:
```
gap> LiePRingsByLibrary(4);
[ <LiePRing of dimension 4 over prime p>,
  <LiePRing of dimension 4 over prime p>,
  <LiePRing of dimension 4 over prime p>,
  <LiePRing of dimension 4 over prime p>,
  <LiePRing of dimension 4 over prime p>,
  <LiePRing of dimension 4 over prime p>,
  <LiePRing of dimension 4 over prime p>,
  <LiePRing of dimension 4 over prime p with parameters [ w ]>,
  <LiePRing of dimension 4 over prime p>,
  <LiePRing of dimension 4 over prime p>,
  <LiePRing of dimension 4 over prime p>,
  <LiePRing of dimension 4 over prime p>,
  <LiePRing of dimension 4 over prime p>,
  <LiePRing of dimension 4 over prime p>,
  <LiePRing of dimension 4 over prime p> ]
```
Now we get this:
```
gap> LiePRingsByLibrary(4);
[ <LiePRing of dimension 4 over prime p>,
  <LiePRing of dimension 4 over prime p>,
  <LiePRing of dimension 4 over prime p>,
  <LiePRing of dimension 4 over prime p>,
  <LiePRing of dimension 4 over prime p>,
  <LiePRing of dimension 4 over prime p>,
  <LiePRing of dimension 4 over prime p>,
  <LiePRing of dimension 4 over prime p>,
  <LiePRing of dimension 4 over prime p>,
  <LiePRing of dimension 4 over prime p>,
  <LiePRing of dimension 4 over prime p>,
  <LiePRing of dimension 4 over prime p>,
  <LiePRing of dimension 4 over prime p>,
  <LiePRing of dimension 4 over prime p>,
  <LiePRing of dimension 4 over prime p> ]
```

There are more examples which can be seen by running the test suite (which also shows tons of other diffs, but those are mostly because printing in general. E.g. for order p^6, the description of number 63 changed from

    [ "all x,y, [x,y]~[x',y'] if y^2-wx^2=y'^2-wx'^2 mod p", "" ]

to

    [ "[x,y]~[x',y'] if y^2-wx^2=y'^2-wx'^2", "" ]

Which is fine -- we can easily "fix" the tests by let GAP rewrite them for us (I can show you how, or else do it for you, but let's first get the actual bugs sorted out)


> >     gap> K := SpecialisePrimeOfLiePRing(L, 7);
> >     <LiePRing of dimension 6 over prime 7>
> >     gap> b := BasisOfLiePRing(K);
> >     [ l1, l2, l3, l4, l5, l6 ]
> >     gap> U := LiePSubring(L, [5*b[1]]);
> >     <LiePRing of dimension 1 over prime p>
>
> Ja, das ist ein Fehler im Manual. U sollte ein Subring von K sein, nicht von L.

OK, good. Once the manual example is fixed, the test file needs ot be


> >     gap> grps := GroupsViaLiePRings(5,5);;
> >     gap> Length(grps) = NrSmallGroups(5^5);
> >     true
> >     gap> Set(GroupsViaLiePRings(5,5),IdGroup) = List([1..77],i->[5^5,i]);
> >     true
> > has changed and now reports `false` in both cases.
>
> Das kann ich nicht reproduzieren.

Interesting. I definitely get this with the LiePRing development version in git:
```
gap> grps := GroupsViaLiePRings(5,5);;
gap> Length(grps);
76
gap> NrSmallGroups(5^5);
77
gap> Difference(List([1..77],i->[5^5,i]), Set(GroupsViaLiePRings(5,5),IdGroup));
[ [ 3125, 1 ] ]
```


> Ausserdem möchte ich das LiePRing Package erweitern um die LiePRings der Ordnung pˆ8 von maximaler Klasse. Ich vermute, ich sollte die Development-Version von GAP per GitHub herunterladen, dort Änderungen machen und dann wieder pushen? Ist das der Dienstweg für solche Probleme? Viele Grüsse, Bettina

Absolutely, that's what you should do, ideally. And if you ping me, I can help ensure the test suite runs again, and that we get a clean release passing all tests.

**Comment by fingolfin** [comment]
Resolved in 2.6
